src/io/flutter/actions/FlutterSdkAction.java

```diff
@@ -88,11 +85,14 @@ public void actionPerformed(AnActionEvent event) {
   public final void perform(@NotNull FlutterSdk sdk,
                             @NotNull Project project,
                             @Nullable AnActionEvent event,
-                            @SuppressWarnings("SameParameterValue") boolean logAction)
+                            @SuppressWarnings("SameParameterValue") boolean sendAnalytics)
     throws ExecutionException {
-    if (logAction) {
-      sendActionEvent();
+    if (sendAnalytics) {
+      FlutterInitializer.sendAnalyticsAction(this);
     }
+
+    FileDocumentManager.getInstance().saveAllDocuments();
```

Comment:
Take a look at the implementation, it takes write action inside if needed, so we are fine here because `actionPerformed()` is called in EDT.